app/services/oidc_jwt_token.rb

```diff
@@ -0,0 +1,8 @@
+class OidcJwtToken < JwtToken
+  def decode
+    return if token.blank?
+    raise JWT::DecodeError unless OidcJwtTokenValidate.new(token).valid_jwt?
```

Comment:
In `valid_jwt?` you catch the `JWT::DecodeError` and return false. Here you raise another. That feels a bit odd to me. You're also decoding twice: once in the validate and if it's present. Then you discard that. Am I missing something or should these be combined?